packages/flame/lib/src/components/position_component.dart

```diff
@@ -1,218 +1,333 @@
+  /// coordinates.
+  Vector2 toLocal(Vector2 point) => _transform.globalToLocal(point);
+
+  /// Transform [point] from the global (world) coordinate space into the
+  /// local coordinates. This can be used, for example, to detect whether
+  /// a specific point on the screen lies within this [PositionComponent],
+  /// and where exactly it is.
+  Vector2 absoluteToLocal(Vector2 point) {
+    var c = parent;
+    while (c != null) {
+      if (c is PositionComponent) {
+        return toLocal(c.absoluteToLocal(point));
+      }
+      c = c.parent;
+    }
+    return toLocal(point);
+  }
 
-  /// Angle (with respect to the x-axis) this component should be rendered with.
-  /// It is rotated around its anchor.
-  double angle;
+  /// The top-left corner's position in the parent's coordinates.
+  Vector2 get topLeftPosition => positionOfAnchor(Anchor.topLeft);
+  set topLeftPosition(Vector2 point) {
+    position += point - topLeftPosition;
+  }
 
-  /// Anchor point for this component. This is where flame "grabs it".
-  /// The [position] is relative to this point inside the component.
-  /// The [angle] is rotated around this point.
-  Anchor anchor;
+  /// The position of the center of the component's bounding rectangle
+  /// in the parent's coordinates.
+  Vector2 get center => positionOfAnchor(Anchor.center);
+  set center(Vector2 point) {
+    position += point - center;
+  }
 
-  /// Whether this component should be flipped on the X axis before being rendered.
-  bool renderFlipX = false;
+  /// The [anchor]'s position in absolute (world) coordinates.
+  Vector2 get absolutePosition => absolutePositionOfAnchor(_anchor);
 
-  /// Whether this component should be flipped ofn the Y axis before being rendered.
-  bool renderFlipY = false;
+  /// Get the absolute top left position regardless of whether it is a child or not
+  Vector2 get absoluteTopLeftPosition =>
+      absolutePositionOfAnchor(Anchor.topLeft);
 
-  PositionComponent({
-    Vector2? position,
-    Vector2? size,
-    Vector2? scale,
-    this.angle = 0.0,
-    this.anchor = Anchor.topLeft,
-    this.renderFlipX = false,
-    this.renderFlipY = false,
-    int? priority,
-  })  : _position = position ?? Vector2.zero(),
-        _size = size ?? Vector2.zero(),
-        _scale = scale ?? Vector2.all(1.0),
-        super(priority: priority);
+  /// Get the absolute center of the component
+  Vector2 get absoluteCenter => absolutePositionOfAnchor(Anchor.center);
 
-  @override
-  bool containsPoint(Vector2 point) {
-    final rectangle = Rectangle.fromRect(toAbsoluteRect(), angle: angle)
-      ..position = absoluteCenter;
-    return rectangle.containsPoint(point);
+  //----------------------------------------------------------------------------
+  // Mutators
+  //----------------------------------------------------------------------------
+
+  /// Flip the component horizontally around its anchor point.
+  void flipHorizontally() => _transform.flipHorizontally();
```

Comment:
I think these two could cause some confusion, because gestures will not be handled where the component is rendered with this right? Since it could be flipped outside of its bounds when the anchor is something else than `center`?
If that is the case, I think we should remove these two and let the `flip*AroundCenter` take their names.

Reply:
Yes, so I was actually thinking about this, and I believe it would likely depend on a specific use case.

For example, one situation when image flipping may be used is when you have a character in a side-view game, and that character is normally going left-to-right, but then you want to go back (right-to-left) and you flip the sprite then. Now, we need to pick a character sprite whose anchor is off-center.  For example, this (the small [+] indicates the anchor):
![mouse](https://user-images.githubusercontent.com/4231472/129623781-e605afa3-ee08-4cce-8b3e-2889723a8f91.png)

Now your mouse is sitting on a floor, and you want it to go the other direction. How you would prefer to flip it:
![mice](https://user-images.githubusercontent.com/4231472/129625578-f782723a-bd09-4cf7-bcd9-5812eb62915f.png)
To me, the second option is much better, since the body of the mouse remains visually in place. 

Same goes for vertical flipping. Right now the mouse is floor-bound, so her anchor is on the floor. Now imagine she can climb the ceilings too: I would want to flip the image vertically for that. However, when she's climbing the ceiling, I would want her anchor to be one ceiling, and the body of the mouse below the anchor -- thus, the flip should occur around the anchor, not around the center line.

What is your thinking here, when would flipping around the center be preferable?

As for your concerns that this would interfere with tap detection -- it shouldn't, as long as gesture detectors use `PositionComponent`'s own `containsPoint()`, or when implementing their own, use the provided `toLocal()` method. The `Transform2D` object handles all these complexities in one centralized location, which was the main point of that class.

Reply:
Aah, that's great! Didn't get that `toLocal` took care of that case, but it's obvious now when you've explained it.